scripts/code_coverage_report/generate_code_coverage_report/Sources/IncrementalCoverageReportGenerator/main.swift

```diff
@@ -0,0 +1,199 @@
+  let file: String
+  let addedLines: [Int]
+  enum CodingKeys: String, CodingKey {
+    case file
+    case addedLines = "added_lines"
+  }
+}
+
+extension FileIncrementalChanges {
+  static func dataModel(_ filePath: String) throws -> [FileIncrementalChanges] {
+    guard let data = JSONParser.readJSON(of: [FileIncrementalChanges].self, from: filePath)
+    else { throw ValidationError("Badly data transform.") }
+    return data
+  }
+}
+
+struct LineCoverage: Codable {
+  var fileName: String
+  var coverage: [Int?]
+  var xcresultBundle: String
+}
+
+struct IncrementalCoverageReportGenerator: ParsableCommand {
+  @Option(
+    help: "Root path of archived files in a xcresult bundle."
+  )
+  var fileArchiveRootPath: String
+
+  @Option(help: "A dir of xcresult bundles.")
+  var xcresultDir: String
+
+  @Option(
+    help: """
+    A JSON file with changed files and added line numbers. E.g. JSON file:
+    '[{"file": "FirebaseDatabase/Sources/Api/FIRDataSnapshot.m", "added_lines": [105,106,107,108,109]},{"file": "FirebaseDatabase/Sources/Core/Utilities/FPath.m", "added_lines": [304,305,306,307,308,309]}]'
+    """,
+    transform: FileIncrementalChanges.dataModel
+  )
+  var changedFiles: [FileIncrementalChanges]
+
+  @Option(
+    help: "Uncovered line JSON file output path"
+  )
+  var uncovered_line_file_json: String = Constants.DEFAULT_UNCOVERED_LINE_REPORT_FILE_NAME
+
+  func readFile(_ fileURL: URL) -> String {
+    var fileContent = ""
+    do {
+      fileContent = try String(contentsOf: fileURL)
+    } catch {
+      print("Failed reading from URL: \(fileURL), Error: " + error.localizedDescription)
+    }
+    return fileContent
+  }
+
+  // This will transfer line executions report from a xcresult bundle to an array of LineCoverage.
+  // The output will have the file name, line execution counts and the xcresult bundle name.
+  func createLineCoverageRecord(from coverageFile: String, changedFile: String,
+                                lineCoverage: [LineCoverage]? = nil) -> [LineCoverage] {
+    // The indices of the array represent the (line index - 1), while the value is the execution counts.
+    var lineExecutionCounts: [Int?] = []
+    if let dir = URL(string: "file://" + FileManager.default.currentDirectoryPath) {
+      let fileURL = dir.appendingPathComponent(coverageFile)
+      let inString = readFile(fileURL)
+      let lineCoverageRegex = try! NSRegularExpression(pattern: Constants
+        .LINE_EXECUTION_COUNT_PATTERN)
+      for line in inString.components(separatedBy: "\n") {
+        let range = NSRange(location: 0, length: line.utf16.count)
+        if let match = lineCoverageRegex.firstMatch(in: line, options: [], range: range) {
+          // Get the execution counts and append. Line indices are
+          // consecutive and so the array will have counts for each line
+          // and nil for unexecutable lines, e.g. comments.
+          let nsRange = match.range(at: 2)
```

Comment:
Where does the `2` come from? Can you provide an example comment above what the line would look like?